camera_app.py

```diff
@@ -204,145 +210,139 @@ def save_photo(img, aux_text):
+                        # If the face is still, start or continue the stillness timer
+                        if face_still_start_time is None:
+                            face_still_start_time = time.time()
+                    else:
+                        # If the face is moving, reset the stillness timer
+                        face_still_start_time = None
+                
+                # Update the last position for the next frame's calculation
+                last_face_position = current_face_center
+                
+                
+                # Check if the face has been still for over a second
+                if face_still_start_time is not None and (time.time() - face_still_start_time > 1.0):
+                    # Check if enough time has passed since the LAST command to avoid spamming
+                    if time.time() - last_guidance_time > 3:
+                        # SUCCESS CONDITION 
+                        if is_in_position and is_face_straight:
+                            textToSpeech("Perfect, hold still!")
+                            save_photo(original_frame_for_photo, target_command)
+                            textToSpeech("Photo taken!")
+                            break
+                        
+                        # GUIDANCE LOGIC 
+                        guidance_message = ""
+                        if not is_in_position:
+                             if current_face_center[1] < target_rect[1]: guidance_message += "Move down. "
+                             elif current_face_center[1] > target_rect[3]: guidance_message += "Move up. "
+                             if current_face_center[0] < target_rect[0]: guidance_message += "Move to your right. "
+                             elif current_face_center[0] > target_rect[2]: guidance_message += "Move to your left. "
+                        elif not is_face_straight: guidance_message = "Please level your head."
+                        elif len(eyes) < 2: guidance_message = "Please face the camera."
+                        
+                        if guidance_message:
+                            textToSpeech(guidance_message)
+                            last_guidance_time = time.time()
+                            # Reset stillness timer after speaking to wait for the next stop
+                            face_still_start_time = None 
+            else:
+                # If the face is lost from the frame, reset the tracking variables
+                last_face_position = None
+                face_still_start_time = None
+                if time.time() - last_guidance_time > 5:
+                    textToSpeech("I can't see your face.")
+                    last_guidance_time = time.time()
+            
 
         cv2.imshow("Selfie Helper", video_frame)
-        if cv2.waitKey(1) & 0xFF == ord("q"):
-            break
+        if cv2.waitKey(1) & 0xFF == ord("q"): break
 
-    # CLEANUP
+    if stop_listening:
```

Comment:
should it still be listening for  a command? Can't we shut off listening after we get a command?